src/stateful-set.ts

```diff
@@ -127,16 +130,44 @@ export class StatefulSet extends workload.Workload {
       metadata: props.metadata,
       spec: Lazy.any({ produce: () => this._toKube() }),
     });
-    this._service = props.service;
+    this.service = props.service ?? this.createHeadlessService();
```

Comment:
If you accept the other suggestion, then here's a follow up:

```suggestion
    this.service = props.service ?? this._createHeadlessService();
```